bndtools.core/src/bndtools/central/Central.java

```diff
@@ -239,11 +238,21 @@ public synchronized static Workspace getWorkspace() throws Exception {
         }
     }
 
-    public synchronized static void onWorkspaceInit(Function<Workspace,Void> callback) {
-        if (workspace != null)
-            callback.run(workspace);
-        else
-            workspaceInitCallbackQueue.add(callback);
+    public static void onWorkspaceInit(Function<Workspace,Void> callback) {
+        callbacks.add(callback);
+        notifyCallbacks();
+    }
+
+    private static void notifyCallbacks() {
+        synchronized (instance) {
```

Comment:
instance could be null if this method is called from onWorkspaceInit before Central is inited. I would suggest to remove the synchronized(instance) and just make the workspace field volatile.